deltachat-ios/Controller/WebViewViewController.swift

```diff
@@ -192,6 +202,47 @@ class WebViewViewController: UIViewController, WKNavigationDelegate {
         webView.evaluateJavaScript("WKWebView_SearchPrev()", completionHandler: nil)
         updateAccessoryBar()
     }
+
+    func openChatFor(url: URL) {
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate,
+              let emailAddress = parseEmailAddress(from: url) else {
+            return
+        }
+
+        // FIXME: lookupContactIdByAddress is still broken
```

Comment:
what precisely is broken?

Reply:
/me suggesting to add these information to the code comment :)

Reply:
from what I tested, ` dcContext.lookupContactIdByAddr(addr)` just always returns 1.  Exact the same work-around has been used on iOS in a different spot already, that's why 'still'.

Reply:
hm, we should file an issue for that, from core-side, that should work.

android uses `lookupContactIdByAddr` as well and checks if a chat with that contact exists (which is more accurate than checking if the contact exist in the addressbook):

https://github.com/deltachat/deltachat-android/blob/master/src/org/thoughtcrime/securesms/NewConversationActivity.java#L156